DBZ-6592 Documenting database permissions required by the MongoDB connector (#4660)
debezium/debezium: documentation/modules/ROOT/pages/connectors/mongodb.adoc

For more information, see the https://docs.mongodb.com/manual/core/replica-set-architectures/[MongoDB documentation].
====

// Type: concept
// ModuleID: mongodb-user-permissions-required-by-debezium-connectors
// Title: User permissions required by {prodname} connectors
[[required-user-permissions]]
=== Required user permissions
Depending on the value of xref:mongodb-property-capture-scope[`capture.scope`] property the user used by the connector needs either the permissions to read any database (when xref:mongodb-property-capture-scope[`capture.scope`] is set to `deployment`) or the permissions to read the database specified by xref:mongodb-property-capture-target[`capture.target`] (when xref:mongodb-property-capture-scope[`capture.scope`] is set to `database`).

> In any case the user should also have the permission to execute the https://www.mongodb.com/docs/manual/reference/command/ping/[ping] command.

Reviewer: ```suggestion
.Permission to use the MongoDB `ping` command 

Regardless of the `capture.scope` setting, the user requires permission to run the MongoDB https://www.mongodb.com/docs/manual/reference/command/ping/[ping] command.
```

> Additionally when capturing changes from a sharded MongoDB cluster and the xref:mongodb-property-mongodb-connection-mode[`mongodb.connection.mode`] property is set to `replica_set` the user used by the connector also needs the permission to read the `config.shards` system collection.

Reviewer: ```suggestion
.Permission to read the `config.shards` collection

For connectors that capture changes from a sharded MongoDB cluster, and for which the xref:mongodb-property-mongodb-connection-mode[`mongodb.connection.mode`] property is set to `replica_set`, you must grant the user permission to read the `config.shards` system collection.
```

To match the name of a schema, {prodname} applies the regular expression that you specify as an _anchored_ regular expression.
That is, the specified expression is matched against the entire name string of the schema; it does not match substrings that might be present in a schema name.

|[[mongodb-property-capture-scope]]<<mongodb-property-capture-scope, `+capture.scope+`>>
|`deployment`

> |Specifies the https://www.mongodb.com/docs/manual/changeStreams/#watch-a-collection--database--or-deployment[scope of change streamed] opened by the connector.

Reviewer: ```suggestion
|Specifies the https://www.mongodb.com/docs/manual/changeStreams/#watch-a-collection--database--or-deployment[scope of the change streams] that the connector opens.
```

Set this property to one of the following values:

> `deployment`:: opens a change stream cursor for a deployment (either a replica set or a sharded cluster) to watch for changes to all non-system collections across all databases except for `admin`, `local`, and `config`.

Reviewer: ```suggestion
`deployment`:: Opens a change stream cursor for a deployment (either a replica set or a sharded cluster) to watch for changes to all non-system collections across all databases, except for `admin`, `local`, and `config`.
```

> `database`:: opens a change stream cursor for a single database to watch for changes to all its non-system collections.

Reviewer: ```suggestion
`database`:: Opens a change stream cursor for a single database to watch for changes to all of its non-system collections.
```

+
[WARNING]
====

> To use signaling when `database` is specified as the value the signaling collection needs to be located in the database specified by the xref:mongodb-property-capture-target[`capture.target`] property.

Reviewer: ```suggestion
If you set the value of this property to `database`,  to use {prodname} signaling , the signaling collection must reside in the database that is  specified by the xref:mongodb-property-capture-target[`capture.target`] property.
```

====

|[[mongodb-property-capture-target]]<<mongodb-property-capture-target, `+capture.target+`>>
|

> | Specifies the database to watch for changes when xref:mongodb-property-capture-scope[`capture.scope`] is set to `database`

Reviewer: ```suggestion
| Specifies the database that the connector monitors for changes.
This property applies only if the xref:mongodb-property-capture-scope[`capture.scope`] is set to `database`.
```

To match the name of a schema, {prodname} applies the regular expression that you specify as an _anchored_ regular expression.
That is, the specified expression is matched against the entire name string of the schema; it does not match substrings that might be present in a schema name.

|[[mongodb-property-capture-scope]]<<mongodb-property-capture-scope, `+capture.scope+`>>
|`deployment`
|Specifies the https://www.mongodb.com/docs/manual/changeStreams/#watch-a-collection--database--or-deployment[scope of the change streams] that the connector opens.
Set this property to one of the following values:

`deployment`:: Opens a change stream cursor for a deployment (either a replica set or a sharded cluster) to watch for changes to all non-system collections across all databases, except for `admin`, `local`, and `config`.

`database`:: Opens a change stream cursor for a single database to watch for changes to all of its non-system collections.
+
[WARNING]
====

> If you set the value of this property to `database`,  to use {prodname} signaling , the signaling collection must reside in the database that is  specified by the xref:mongodb-property-capture-target[`capture.target`] property.

Reviewer: ```suggestion
If you set the value of this property to `database`, to use {prodname} signaling, you the signaling collection that you create must reside in the database that is specified by the xref:mongodb-property-capture-target[`capture.target`] property.
```
Removed some extra spaces and slightly reworded.
